scripts/download-browsers.js

```diff
@@ -0,0 +1,135 @@
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+require("dotenv").config();
+
+const util = require("util");
+const path = require("path");
+const {
+  downloadBrowserWithProgressBar,
+} = require("playwright/lib/utils/browserFetcher");
+const SYNTHETICS_BROWSER_REVISIONS = require("@elastic/synthetics/node_modules/playwright-chromium/browsers.json");
+const EXECUTABLE_PATHS = {
+  chromium: {
+    linux: ["chrome-linux", "chrome"],
+    mac: ["chrome-mac", "Chromium.app", "Contents", "MacOS", "Chromium"],
+    win: ["chrome-win", "chrome.exe"],
+  },
+  ffmpeg: {
+    linux: ["ffmpeg-linux"],
+    mac: ["ffmpeg-mac"],
+    win: ["ffmpeg-win64.exe"],
+  },
+};
+
+const DOWNLOAD_URLS = {
+  chromium: {
+    linux: "%s/builds/chromium/%s/chromium-linux.zip",
+    mac: "%s/builds/chromium/%s/chromium-mac.zip",
+    win: "%s/builds/chromium/%s/chromium-win64.zip",
+    // 'mac-arm64': '%s/builds/chromium/%s/chromium-mac-arm64.zip',
+  },
+  ffmpeg: {
+    linux: "%s/builds/ffmpeg/%s/ffmpeg-linux.zip",
+    mac: "%s/builds/ffmpeg/%s/ffmpeg-mac.zip",
+    win: "%s/builds/ffmpeg/%s/ffmpeg-win64.zip",
+    // 'mac-arm64': '%s/builds/ffmpeg/%s/ffmpeg-mac.zip',
+  },
+};
+
+async function download(platform, browser, revision, directory) {
+  const executablePath = findExecutablePath(directory, browser, platform);
+  const downloadHost =
+    process.env["PLAYWRIGHT_DOWNLOAD_HOST"] ||
+    "https://playwright.azureedge.net";
+  const downloadURL = util.format(
+    DOWNLOAD_URLS[browser][platform],
+    downloadHost,
+    revision
+  );
+  const title = `${browser} v${revision} for ${platform}`;
+  const downloadFileName = `playwright-download-${browser}-${platform}-${revision}.zip`;
+  try {
+    // eslint-disable-next-line no-console
+    console.info("Downloading browser ", title);
+    await downloadBrowserWithProgressBar(
+      title,
+      directory,
+      executablePath,
+      downloadURL,
+      downloadFileName
+    );
+  } catch (e) {
+    throw new Error(`Failed to download ${title}, caused by\n${e.stack}`);
+  }
+}
+
+function findExecutablePath(dir, browser, platform) {
+  const tokens = EXECUTABLE_PATHS[browser][platform];
+  return tokens ? path.join(dir, ...tokens) : undefined;
+}
+
+function getBrowserVersion(browser) {
+  const { browsers } = SYNTHETICS_BROWSER_REVISIONS;
+  const { revision } = browsers.find(br => br.name === browser) || {};
+  return revision;
+}
+
+function translatePlatform(platform) {
+  switch (platform) {
+    case "win32":
+      return "win";
+    case "darwin":
+      return "mac";
+    default:
+      return platform;
+  }
+}
+
+exports.downloadForPlatform = async function downloadForPlatform(platform) {
+  platform = translatePlatform(platform);
+  for (const browser of ["chromium", "ffmpeg"]) {
```

Comment:
Nit: should we change `browser` to `software` here and in the references within the function like `getBrowserVersion` given we're not only dealing with browsers here?

Reply:
That's a good point, will amend 👍